src/ewatercycle/parameter_set.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+from typing import Set
+
+from pydantic import BaseModel
+
+from ewatercycle.util import to_absolute_path
+
+
+class ParameterSet(BaseModel):
+    """Container object for parameter set options."""
+
+    name: str = ""
+    """Name of parameter set"""
+    directory: Path
+    """Location on disk where files of parameter set are stored.
+    If Path is relative then relative to CFG.parameterset_dir."""
+    config: Path
+    """Model configuration file which uses files from
+    :py:attr:`~directory`. If Path is relative then relative to
+    CFG.parameterset_dir."""
+    doi: str = "N/A"
+    """Persistent identifier of parameter set. For a example a DOI
+    for a Zenodo record."""
+    target_model: str = "generic"
+    """Name of model that parameter set can work with."""
+    supported_model_versions: Set[str] = set()
+    """Set of model versions that are
+    supported by this parameter set. If not set then parameter set will be
+    supported by all versions of model"""
```

Comment:
```suggestion
    """Set of model versions that are compatible with this parameterset."""
```

I'd love to also rename the setting to `compatible_model_versions`, but that might break stuff.

Reply:
I am not sure about this suggestion.
With your suggestion how would a model developer know what the behavior is of an empty set. An empty set could also mean zero versions are supported/compatible.

Reply:
We could make it optional so 
* None means compatible with all versions 
* set() means no versions are compatible

Reply:
My concern was more about the word "supported" than the behavior for not setting it. Sorry if that was unclear. With respect to making it optional, perhaps we should reconsider this behaviour altogether. Imagine I make a new (backward compatible) release of the model. Then I'd like for my existing parameter sets to work out of the box.